src/cli.js

```diff
@@ -32,9 +38,15 @@ export function run(args) {
 		config.coverage.cleanDir = commander.cleanCoverageDir;
 	}
 
-	jsApi.runTests(config, function(err) {
+	jsApi.runTests(config, function(err, safeExit) {
 		console.error(err);
-		process.exit(1);
+		if (!isWatching()) {
+			if (safeExit) {
+				safeExit();
+			} else {
+				process.exit(1);
```

Comment:
@joeldenning Also, I think you might want to use [this](https://www.npmjs.com/package/exit) instead of explicitly calling `process.exit(1);`.